lib/variable/variable_instantiate_linalg.cpp

```diff
@@ -42,13 +53,16 @@ std::vector<std::string> element_keys(const Variable &var) {
     return {"x", "y", "z"};
   if (variableFactory().elem_dtype(var) == dtype<Eigen::Matrix3d>)
     return {"xx", "xy", "xz", "yx", "yy", "yz", "zx", "zy", "zz"};
+  if (variableFactory().elem_dtype(var) == dtype<Eigen::Affine3d>)
+    throw except::TypeError("Not supported for Affine3d types");
   if (variableFactory().elem_dtype(var) == dtype<scipp::index_pair>)
     return {"begin", "end"};
   throw except::TypeError("dtype is not structured");
 }
 
 INSTANTIATE_STRUCTURE_ARRAY_VARIABLE(vector_3_float64, Eigen::Vector3d, double)
 INSTANTIATE_STRUCTURE_ARRAY_VARIABLE(matrix_3_float64, Eigen::Matrix3d, double)
+INSTANTIATE_STRUCTURE_ARRAY_VARIABLE(transform, Eigen::Affine3d, double)
```

Comment:
Have we reached a conclusion regarding naming?

Reply:
Happy to change the name - do you have a better suggestion?

Reply:
Under the assumption that we will also need linear transforms, I think our options are:

1.
```python
linear_transform
affine_transform
```

2.
```python
transform
affine_transform
```

3.
```python
linear_transform
transform
```

But maybe the question is: Do we need linear transforms, or just we support pure rotation and pure scaling, but any combination of the two is represented as an affine transform? I think this is actually not possible, since a affine transform requires a translation unit, but rotation and scaling do not have one. That is, we need a dtype other than affine transform to represent `rotation * scaling*`.

So back to the initial question: 1., 2., or 3.?

Reply:
Option 1 seems to be the least ambiguous to me. I've changed it.